src/jquery.multiselect.js

```diff
@@ -90,110 +140,130 @@
-              'id': elSelect.id ? elSelect.id  + '_ms' : null
-            })
-            .prop('aria-haspopup', true)
-            .html('<span class="ui-multiselect-open">' + iconSet.open + '</span>');    // Necessary to simplify dynamically changing the open icon.
-
-      this.$buttonlabel = $( document.createElement('span') )
-            .html(options.noneSelectedText || $element[0].placeholder)
-            .appendTo( $button );
-
-      // Header controls, will contain the check all/uncheck all buttons
+      if (!options.listbox) {
+         // The button that opens the widget menu.  Note that this is inserted later below.
+         var $button = ( this.$button = $( document.createElement('button') ) )
+               .addClass('ui-multiselect ui-widget ui-state-default ui-corner-all'
+                           + (/\bbutton\b/i.test(wrapText) ? '' : ' ui-multiselect-nowrap')
+                           + (classes ? ' ' + classes : '')
+                           )
+               .attr({
+                 'type': 'button',
+                 'title': elSelect.title,
+                 'tabIndex': elSelect.tabIndex,
+                 'id': elSelect.id ? elSelect.id  + '_ms' : null
+               })
+               .prop('aria-haspopup', true)
+               .html( _linkHTML('<span class="{{class}}" title="{{title}}">{{icon}}</span>', 'open') );
+
+         this.$buttonlabel = $( document.createElement('span') )
+               .html(options.noneSelectedText || $element[0].placeholder)
+               .appendTo( $button );
+      }
+
+      // Header controls will contain the links & ordering specified by the header option.
       // Depending on how the options are set, this may be empty or simply plain text
-      var headerLinksHTML = ( headerOn === true
-               ?  (checkAllText === null ? '' : '<li><a class="ui-multiselect-all" title="Check All">' + iconSet.checkAll + (checkAllText ? '<span>' + checkAllText + '</span>' : '' ) + '</a></li>')
-                  + (uncheckAllText === null ? '' : '<li><a class="ui-multiselect-none" title="Uncheck All">' + iconSet.uncheckAll + (uncheckAllText ? '<span>' + uncheckAllText + '</span>' : '' ) + '</a></li>')
-                  + (flipAllText === null ? '' : '<li><a class="ui-multiselect-flip" title="Flip All">' + iconSet.flipAll + '<span>' + (flipAllText ? '<span>' + flipAllText + '</span>' : '' ) + '</a></li>')
-               :  (typeof headerOn === 'string' ? '<li>' + headerOn + '</li>' : '') );
+      var headerLinksHTML = '';
+      if (options.header === false) {}  // no-op for no header
+      else if (options.header.charAt(0) === '=') {
+         headerLinksHTML = '<li>' + options.header.slice(1) + '</li>';
+      }
+      else {
+         var hdrLinkTemplate = '<li><a class="{{class}}" title="{{title}}">{{icon}}<span>{{text}}</span></a></li>',
+               headerLinks = options.header
+                              .replace(/all\b/g,'All')
```

Comment:
I really don't like that this implementation of headers requires all this string processing/transformation, and a new syntax for `=`. I'd prefer to address the header functionality separately.

This is too much regex and replace on what was an otherwise simple option and the transform to `All` makes me think it has some impact on translations as well.

I'd be a lot happier with something like a mapping object like
```
{
  checkAll: true
  uncheckAll: true
}
```

Reply:
@mlh758 Ok, so let's unpack this...

> I really don't like that this implementation of headers requires all this string processing/transformation, and a new syntax for =. I'd prefer to address the header functionality separately.

Ok, I wanted to reduce the number of options.   The current header option strikes me as sort of a seldom used option, because the "false" setting is unlikely to be used much.  That is why I thought maybe I could just have values of false and "everything else that implies true" for that option.

>This is too much regex and replace on what was an otherwise simple option and the transform to All makes me think it has some impact on translations as well.

Ok, but you have to keep in mind that I have added three new header link types over what is in the master branch.  I purposely tried to generalize this to avoid continuing down the path of a massive string concatenation operation that would be a big mess.  My thinking is that devs will not likely use all of the header link types but they may want to also specify the order, particularly for a RTL case.

Also, these are just keys for the linkInfo object lookup.   The actual text that would be translated is in the object _values_.

> I'd be a lot happier with something like a mapping object like

I considered that, but you have to keep in mind that unlike arrays, the order of object keys is not guaranteed.   Frequently, they are kept in an internally sorted state to allow for quick look-ups by the JS interpreter.

Because I specifically wanted to allow specifying the order of the header links, I was left with either using an array of keys (for the linkInfo object) or a more convenient comma delimited string of the same that would have to be split--an object is not suitable for this.  The inclusion of a linkInfo key in the array or string implies True, so no object is needed anyway.  The "All" replacement is simply to prevent developer typos from creating issues... since the keys for linkInfo are case sensitive.

I am open to beating this up further, but I am committed to some way of specifying link order to facilitate a future RTL feature and to allow for developer preferences.

Reply:
Adding a sequence field would allow you to still maintain order. Although just saying that it sounds awkward too.

How much of this regex operation is meant to correct for developer typos? It seems like you should just be able to split on commas, map trim over the results, and then continue.

Something like:
```
var headerLinks = options.header.split(',').map(function(headerOption) {
  return headerOption.trim();
}).filter(function(headerOption) {
  if (/open|close|collapse/.test(headerOption)) {
    return false;
  }
  if (options.maxSelected && headerOption === 'checkAll') {
    return false;
  }
  return true;
});
```
It's a little more code, but it's simple and easy to follow, and then the for loop below can just run over the remaining items, removing some complexity from the if check.

Reply:
@mlh758 Interesting use of filter() w/ map()  ... let me dig into that a bit more later.

The "developer oops" stuff is correcting the "All" case, white space stripping, and extra comma elimination.  I can take that out.

Reply:
Ok, I thought you were using jQuery filter & map functions, but I realized later you are using plain JS array functions.

map & filter are IE9+... _Are you ok w/ that?_

Reply:
If you want IE8, also, then $.grep & $.map will have to be used.

Reply:
I specified IE10+ as a requirement when I released 2.0.